contracts/BootstrapRewards.sol

```diff
@@ -0,0 +1,139 @@
+import "./spec_interfaces/IBootstrapRewards.sol";
+
+contract BootstrapRewards is IBootstrapRewards, Ownable {
+    using SafeMath for uint256;
+
+    IContractRegistry contractRegistry;
+
+    uint256 pool;
+
+    // TODO - should the fixed pool rate be a function of the committee size?
+    uint256 poolMonthlyRate; // todo - deprecate this
+    uint256 generalCommitteeMonthlyRate; // todo - apply this rate
+    uint256 complianceCommitteeMonthlyRate; // todo - apply this rate
+
+    uint256 lastPayedAt;
+
+    mapping(address => uint256) externalTokenBalance;
+
+    IERC20 externalToken;
+    address rewardsGovernor;
+
+    modifier onlyRewardsGovernor() {
+        require(msg.sender == rewardsGovernor, "caller is not the rewards governor");
+
+        _;
+    }
+
+    constructor(IERC20 _externalToken, address _rewardsGovernor) public {
+        require(address(_externalToken) != address(0), "externalToken must not be 0");
+
+        externalToken = _externalToken;
+        lastPayedAt = now;
+        rewardsGovernor = _rewardsGovernor;
+    }
+
+    function setContractRegistry(IContractRegistry _contractRegistry) external onlyOwner {
+        require(address(_contractRegistry) != address(0), "contractRegistry must not be 0");
+        contractRegistry = _contractRegistry;
+    }
+
+    // todo - deprecate this, use setGeneralCommitteeBootstrapMonthlyRate and setComplianceCommitteeBootstrapMonthlyRate instead
+    function setPoolMonthlyRate(uint256 rate) external onlyRewardsGovernor {
+        _assignRewards();
+        poolMonthlyRate = rate;
+    }
+
+    function setGeneralCommitteeBootstrapMonthlyRate(uint256 rate) external {
+        _assignRewards();
+        generalCommitteeMonthlyRate = rate;
+    }
+
+    function setComplianceCommitteeBootstrapMonthlyRate(uint256 rate) external {
+        _assignRewards();
+        complianceCommitteeMonthlyRate = rate;
+    }
+
+    function topUpBootstrapPool(uint256 amount) external {
+        pool = pool.add(amount);
+        require(externalToken.transferFrom(msg.sender, address(this), amount), "Rewards::topUpFixedPool - insufficient allowance");
+        emit BootstrapAddedToPool(amount, pool);
+    }
+
+    function getBootstrapBalance(address addr) external view returns (uint256) {
+        return externalTokenBalance[addr];
+    }
+
+    function getLastBootstrapAssignment() external view returns (uint256) {
+        return lastPayedAt;
+    }
+
+    function assignRewards() external {
+        _assignRewards();
+    }
+
+    function _assignRewards() private {
+        // TODO we often do integer division for rate related calculation, which floors the result. Do we need to address this?
+        // TODO for an empty committee or a committee with 0 total stake the divided amounts will be locked in the contract FOREVER
+
+        uint256 duration = now.sub(lastPayedAt);
+
+        uint256 amount = Math.min(poolMonthlyRate.mul(duration).div(30 days), pool);
```

Comment:
uses poolMonthlyRate and instead of generalCommitteeMonthlyRate

Reply:
`30 days` in solidity is the number of seconds in 30 days, which is 2592000. Why 86400?

Reply:
sec in a day. can ignore